App/src/main/java/vut/fit/gja2023/app/service/CourseService.java

```diff
@@ -110,6 +116,58 @@ public void addStudentToCourse(@NotNull CourseBo course, @NotNull UserBo student
             .filter(assignment -> !assignment.isTeam())
             .forEach(assignment -> projectService.createProject(student, assignment));
     }
+    
+    public void upsertTeams(@NotNull CourseBo course, @NotNull Map<String, List<String>> teamsStudents) {
+        
+        teamsStudents.forEach((teamName, loginsList) -> {
+            TeamBo newTeam = teamService.generateNewTeam(teamName, course);
+            
+            List<UserBo> existingStudents = userRepository.findAllByLogins(loginsList);
+            
+            loginsList.forEach(login -> {
+                
+                UserBo existingStudent = existingStudents.stream()
+                    .filter(student -> login.matches(student.getLogin()))
+                    .findFirst()
+                    .orElse(null);
+
+                //student already exists
+                if (existingStudent != null) {
+                    
+                    // student is not yet attending this course
+                    if (!existingStudent.getStudiedCourses().contains(course)) {
+                        addStudentToCourse(course, existingStudent);
+                    }
+                    
+                } else {
+                    //student has to be created
+                    UserBo generatedStudent = userService.generateUnnamedUser(login, UserRole.STUDENT);
+                    addStudentToCourse(course, generatedStudent);
+                    
+                    existingStudent = generatedStudent;
+                }
+                
+                addTeamToStudent(existingStudent, newTeam);
+                addStudentToTeam(newTeam, existingStudent);
+            });
```

Comment:
není tady pohlídanej reupload týmů, to znamená když student existuje a už má v daným předmětu jienj tým ... takhle může být ve více týmech zároveň ... tak by bylo dobrý aby se týmy ihned přihlásili na týmový zadání, takže aby se vytvořily týmový projekty a daným týmem a složky týmových projektů